http/vertx/src/test/java/io/quarkus/ts/vertx/ThreadPoolIT.java

```diff
@@ -0,0 +1,90 @@
+package io.quarkus.ts.vertx;
+
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.junit.jupiter.api.Assertions.assertTrue;
+
+import java.util.Arrays;
+
+import org.junit.jupiter.api.Tag;
+import org.junit.jupiter.api.Test;
+
+import io.quarkus.test.bootstrap.Protocol;
+import io.quarkus.test.bootstrap.RestService;
+import io.quarkus.test.scenarios.QuarkusScenario;
+import io.quarkus.test.scenarios.annotations.DisabledOnNative;
+import io.quarkus.test.services.QuarkusApplication;
+import io.quarkus.ts.vertx.AbstractVertxIT.Metric;
+import io.smallrye.mutiny.Uni;
+import io.vertx.core.http.HttpMethod;
+import io.vertx.mutiny.core.Vertx;
+import io.vertx.mutiny.core.http.HttpClientRequest;
+import io.vertx.mutiny.core.http.HttpClientResponse;
+
+@DisabledOnNative // thread pool test is not relevant for a native mode
+@QuarkusScenario
+public class ThreadPoolIT {
+
+    @QuarkusApplication
+    static final RestService app = new RestService()
+            .withProperty("quarkus.thread-pool.core-threads", "1") // set explicitly in case default changes
+            .withProperty("quarkus.thread-pool.max-threads", "1")
+            .withProperty("quarkus.thread-pool.growth-resistance", "1f")
+            .withProperty("quarkus.thread-pool.queue-size", "0");
+
+    @Tag("QUARKUS-3535")
+    @Test
+    public void testRejectedPoolSubmissionMetric() {
+        assertEquals(0, numberOfRejectedRequests());
+        callHelloWorld(1);
+        assertEquals(0, numberOfRejectedRequests());
+        callHelloWorld(10);
+        float numberOfRejected = numberOfRejectedRequests();
+        assertTrue(numberOfRejected > 0);
+    }
+
+    private static void callHelloWorld(int howManyTimes) {
+        var vertx = Vertx.vertx();
+        var httpClient = vertx.createHttpClient();
+        try {
+            var appURI = app.getURI(Protocol.HTTP);
+            var joinBuilder = Uni.join().<Void> builder();
+            for (int i = 0; i < howManyTimes; i++) {
+                var request = httpClient
```

Comment:
What are the default client timeouts? Do we want to adjust them to wait shorter time?

Reply:
By default timeout is disabled - https://vertx.io/docs/apidocs/io/vertx/core/http/RequestOptions.html. I think timeout is not bad idea, although request should be rejected by Quarkus app anyway (which is point of the test), but we can recognize how request got rejected by both metric and 503 response status.

I added timeout, but I'm not sure how will it behave on very weak machines (simply don't know), we might tweak it in case the test become flaky.